tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -72,15 +74,30 @@ public abstract class ResourceTest extends CoreTest {
 	//constants for nature sets
 	protected static final String SET_STATE = "org.eclipse.core.tests.resources.stateSet";
 
+	private static ServiceTracker<IWorkspace, IWorkspace> workspaceTracker;
+
 	/**
 	 * Set of FileStore instances that must be deleted when the
 	 * test is complete
 	 * @see #getTempStore
 	 */
 	private final Set<IFileStore> storesToDelete = new HashSet<>();
 
-	public static IWorkspace getWorkspace() {
-		return ResourcesPlugin.getWorkspace();
+	public static synchronized IWorkspace getWorkspace() {
```

Comment:
Why that? I have no IDE at hand now, but why this trivial getter that is supposed to read a field is replaced with something ... complex and non trivial?

Reply:
I used that to test if it is simply that the workspace is not ready yet, but its more likely cause by #86